core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -800,6 +814,10 @@ private[spark] class TaskSetManager(
     info.markFinished(TaskState.FINISHED, clock.getTimeMillis())
     if (speculationEnabled) {
       successfulTaskDurations.insert(info.duration)
+      inefficientTaskCalculator.foreach { inefficientTask =>
+        inefficientTask.updateTaskProgressThreshold(result)
+        inefficientTask.removeRuningTasksProgressRate(tid)
```

Comment:
I think we can just inline `removeRuningTasksProgressRate` into `updateTaskProgressThreshold` since there's no where else invokes `removeRuningTasksProgressRate`.